packages/storage/src/errors/StorageError.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+	AmplifyError,
+	ConsoleLogger as Logger,
+	ErrorParams,
+} from '@aws-amplify/core';
+
+export class StorageError extends AmplifyError {
+	constructor(params: ErrorParams) {
+		super(params);
+
+		// TODO: Delete the following 2 lines after we change the build target to >= es2015
+		this.constructor = StorageError;
+		Object.setPrototypeOf(this, StorageError.prototype);
+	}
+}
+
+export enum StorageErrorTypes {
+	NO_CREDENTIALS = 'NO_CREDENTIALS',
+}
+
+const StorageErrorStrings: { [key in StorageErrorTypes]: ErrorParams } = {
+	[StorageErrorTypes.NO_CREDENTIALS]: {
+		name: 'NoCredentials',
+		message: 'No credentials',
+		recoverySuggestion: ``,
```

Comment:
```suggestion
		recoverySuggestion: '',
```
But do we need to pass empty `recoverySuggestion` here?